File: Week 1 2 3/drawingBook/DrawingBook.java

package drawingBook;

public class DrawingBook {
    
    public static void main(String[] args){
        int n = 5;
        int p = 4;
        System.out.println(pageCount(n,p));
    }

    public static Number pageCount(int n,int p){
    int fromFront =(int) Math.floor(p / 2);
  
   
    int fromBack = (int) (Math.floor(n / 2) - Math.floor(p / 2));
    return Math.min(fromFront, fromBack);
    }
    
}

File: Week 1 2 3/drawingBook/drawingBook.js
function pageCount(n, p) {
  // Calculamos el número de páginas a voltear desde el principio
  const fromFront = Math.floor(p / 2);

  // Calculamos el número de páginas a voltear desde el final
  const fromBack = Math.floor(n / 2) - Math.floor(p / 2);

  // Devolvemos el mínimo de páginas a voltear
  return Math.min(fromFront, fromBack);
}
const n = 5;
const p = 4;
console.log(pageCount(n, p));
